hummingbot/connector/exchange/ndax/ndax_exchange.py

```diff
@@ -65,24 +76,193 @@ def __init__(self,
+        """
+        Sends an aiohttp request and waits for a response.
+        :param method: The HTTP method, e.g. get or post
+        :param path_url: The path url or the API end point
+        :param is_auth_required: Whether an authentication is required, when True the function will add encrypted
+        signature to the request.
+        :returns A response in json format.
+        """
+        url = CONSTANTS.REST_URL + path_url
+        client = await self._http_client()
+
+        try:
+            if is_auth_required:
+                headers = self._auth.get_auth_headers()
+            else:
+                headers = self._auth.get_headers()
+
+            if method == "GET":
+                response = await client.get(url, headers=headers, params=params)
+            elif method == "POST":
+                response = await client.post(url, headers=headers, data=ujson.dumps(data))
+            else:
+                raise NotImplementedError(f"{method} HTTP Method not implemented. ")
+
+            parsed_response = await response.json()
+        except ValueError as e:
+            self.logger().error(f"{str(e)}")
+            raise ValueError(f"Error authenticating request {method} {url}. Error: {str(e)}")
+        except Exception as e:
+            raise IOError(f"Error parsing data from {url}. Error: {str(e)}")
+        if response.status != 200:
+            raise IOError(f"Error fetching data from {url}. HTTP status is {response.status}. "
+                          f"Message: {parsed_response} "
+                          f"Params: {params} "
+                          f"Data: {data}")
+
+        return parsed_response
+
+    async def _update_balances(self):
+        """
+        Calls REST API to update total and available balances
+        """
+        local_asset_names = set(self._account_balances.keys())
+        remote_asset_names = set()
+
+        params = {
+            "OMSId": 1,
+            "AccountId": self.account_id
+        }
+        account_positions: List[Dict[str, Any]] = await self._api_request(
+            method="GET",
+            path_url=CONSTANTS.ACCOUNT_POSITION_PATH_URL,
+            params=params,
+            is_auth_required=True
+        )
+        for position in account_positions:
+            asset_name = position["ProductSymbol"]
+            self._account_balances[asset_name] = Decimal(str(position["Amount"]))
+            self._account_available_balances[asset_name] = self._account_balances[asset_name] - Decimal(str(position["Hold"]))
+            remote_asset_names.add(asset_name)
+
+        asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+        for asset_name in asset_names_to_remove:
+            del self._account_available_balances[asset_name]
+            del self._account_balances[asset_name]
+
+    async def _update_order_status(self):
+        # Waiting on buy and sell functionality.
+        pass
+
+    async def _status_polling_loop(self):
```

Comment:
I think the event has to be reset at some point. If not the loop will keep running eternally and all the time.